### PR TITLE
Convert deployment strategy from nginx to Node web/api server.

### PR DESCRIPTION
**Pull request by JustinTRoss**
- Created Procfile to give Heroku custom commands for how to run our web server.
- Added serve command to run server.
- Updated server file
    - to scale cluster across machine cores.
    - to serve both static files and api requests.

### CODE REVIEW

File: Procfile [hunk 1]
```diff
@@ -0,0 +1 @@
+web: npm run serve
```
Comment (JustinTRoss) [inline Procfile:1]: This is a Heroku-specific file used to tell Heroku how to run your code. In this case, we are running a web process.

File: package.json [hunk 2]
```diff
@@ -22,7 +22,8 @@
     "start": "react-scripts start",
     "build": "react-scripts build",
     "test": "react-scripts test",
-    "eject": "react-scripts eject"
+    "eject": "react-scripts eject",
+    "serve": "node server.js"
```
Comment (JustinTRoss) [inline package.json:26]: Added a serve command to start server.

File: server.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+const cluster = require("cluster");
+const numCPUs = require("os").cpus().length;
+
+require("dotenv").config();
+
+const PORT = process.env.port || 8080;
+const CLIENT_DISTRIBUTION_DIRECTORY = path.resolve(__dirname, "./build");
+const isDev = process.env.NODE_ENV !== "production";
+
+// Multi-process to utilize all CPU cores.
```
Comment (JustinTRoss) [inline server.js:15]: Creating node cluster on machine to maximize performance and throughput on the machine.

File: server.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  });
+} else {
+  const app = express();
+
+  app.use(cors());
+  app.use(bodyParser.json());
+  app.use(bodyParser.urlencoded({ extended: false }));
+
+  // Priority serve any static files.
+  app.use(express.static(CLIENT_DISTRIBUTION_DIRECTORY));
```
Comment (JustinTRoss) [inline server.js:37]: Tell server how to serve static files.

File: server.js [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+      } else {
+        res.send("Success");
+      }
+      transporter.close();
+    });
+  });
+
+  // All remaining requests return the React app, so it can handle routing.
+  app.get("*", (req, res) => {
+    res.sendFile(path.resolve(CLIENT_DISTRIBUTION_DIRECTORY, "index.html"));
```
Comment (JustinTRoss) [inline server.js:70]: Tell server to direct unrecognized requests to the index.html page, which will enable client-side routing.

**Comment by JustinTRoss** [comment]
We were seeing the 405 issues because we were using the create-react-app buildpack, which is a static file server buildpack, whereas the intended use is as both a file server and an api server. I have removed the buildpack in Heroku and added node buildpack.

File: server.js [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+const bodyParser = require("body-parser");
+const nodemailer = require("nodemailer");
+const cors = require("cors");
+const path = require("path");
+const cluster = require("cluster");
+const numCPUs = require("os").cpus().length;
+
+require("dotenv").config();
+
+const PORT = process.env.port || 8080;
```
Comment (JustinTRoss) [inline server.js]: It has taken me an hour of banging my head against my computer, but I've realized that `process.env.port` actually needs to be `process.env.PORT`. Remember that to save yourself the headache. :)

**Comment by JustinTRoss** [comment]
The site is functionally serving through the node server now. I pushed to Heroku in your footsteps. The email endpoint still does not work, but we're at least able to make post requests now. I would try walking back through somebody's gmail setup step by step.

File: server.js [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+const bodyParser = require("body-parser");
+const nodemailer = require("nodemailer");
+const cors = require("cors");
+const path = require("path");
+const cluster = require("cluster");
+const numCPUs = require("os").cpus().length;
+
+require("dotenv").config();
+
+const PORT = process.env.PORT || 8080;
```
Comment (JustinTRoss) [inline server.js:11]: It has taken me an hour of banging my head against my computer, but I've realized that `process.env.port` actually needs to be `process.env.PORT`. Remember that to save yourself the headache. :)

Comment (Papatiara) [reply server.js:11]: Thank you so much for doing this!

**Pull request merged**